keras/models.py

```diff
@@ -790,40 +790,78 @@ def compile(self, optimizer, loss,
         self.sample_weights = self.model.sample_weights
         self.targets = self.model.targets
 
-    def fit(self, x, y, batch_size=32, epochs=10, verbose=1, callbacks=None,
-            validation_split=0., validation_data=None, shuffle=True,
-            class_weight=None, sample_weight=None, initial_epoch=0, **kwargs):
+    def fit(self, x=None,
+            y=None,
+            batch_size=32,
+            epochs=10,
+            steps_per_epoch=None,
+            verbose=1,
+            callbacks=None,
+            validation_split=0.,
+            validation_data=None,
+            validation_steps=None,
+            shuffle=True,
+            class_weight=None,
+            sample_weight=None,
+            initial_epoch=0,
+            **kwargs):
         """Trains the model for a fixed number of epochs.
 
         # Arguments
-            x: input data, as a Numpy array or list of Numpy arrays
-                (if the model has multiple inputs).
-            y: labels, as a Numpy array.
+            x: Numpy array of training data or tensor which evaluates to a batch.
+                Can be a list if the model has multiple inputs.
+                If all inputs in the model are named,
+                you can also pass a dictionary
+                mapping input names to Numpy arrays or tensors.
+            y: Numpy array of target data or tensor which evaluates to a batch.
+                Can be a list if the model has multiple outputs.
+                If all outputs in the model are named,
+                you can also pass a dictionary
+                mapping output names to Numpy arrays or tensors.
             batch_size: integer. Number of samples per gradient update.
-            epochs: integer, the number of epochs to train the model.
-            verbose: 0 for no logging to stdout,
-                1 for progress bar logging, 2 for one log line per epoch.
-            callbacks: list of `keras.callbacks.Callback` instances.
-                List of callbacks to apply during training.
+                Not required when using tensors as input.
+            steps_per_epoch: Only relevant when using tensors as input.
+                Total number of steps (batches of samples)
+                to evaluate from input tensors before declaring one epoch
+                finished and starting the next epoch. It should typically
+                be equal to the number of unique samples if your dataset
+                divided by the batch size.
```

Comment:
Yet the tfrecords pipe you have in your example does not support a non-fixed batch size, while you are not sure your dataset size is not cleanly divisible by the batch size. Having a super fast data pipe in tf and having mixed-size batch sizes is hard. Optimized Tensorflow models therefore often works with steps, rather than epochs.